### ISSUE TITLE
Cannot login - Safari

### ISSUE THREAD

**Issue by Clausjac** [grafolean/grafolean#18]
Got Grafolean installed without issues and set it to port 8080. Access the webpage and input information for Admin user. Then I get to the login screen and is asked for credentials. Use the admin username and password from the previous page. I am immediately returned to an empty login form... No error message or anything. Can you help?

**Comment by grafolean** [comment]
If you press `F12` and select `Console` tab, do you see any errors?

**Comment by Clausjac** [comment]
Tried switching browser from Safari to Chrome. Chrome works for me but Safari doesn't.

**Comment by grafolean** [comment]
Glad you could make it work, and thanks for reporting this back. I'll see if I can get my hands on Safari somewhere to test this.

**Comment by Clausjac** [comment]
This is what the debug menu tells me - hope it helps:

![123](https://user-images.githubusercontent.com/79799346/109432748-001d0f00-7a0d-11eb-9ef6-21e13c6e3d7d.png)

**Comment by grafolean** [comment]
@Clausjac Thank you for providing the details!

I finally found some time to take a look at this. Unfortunately I don't have access to Safari browser where I could test it, but it looks like Safari is extra careful with Content Security Headers (Chrome is not following the specs here). 

I believe that setting `MQTT_WS_PORT` in `docker-compose.yml` to the correct port should fix it:
```
    - MQTT_WS_PORT=8080
```

Grafolean should then be restarted to apply the settings:
```
$ docker-compose down
$ docker-compose up -d
```

I would appreciate if you have a chance to try it out, but understand if it is not possible. At the very least this issue might help someone else in the future. :slightly_smiling_face:

**Comment by Clausjac** [comment]
Hi,


I tried to do as you suggested but unfortunately, the result is the same.




Login from Chrome is fine, but Safari encounters an error. Here is a screenshot from the Safari Webinspector:


Kind regards

Claus




On 9 April 2021 at 18:42, grafolean ***@***.***> wrote:






@Clausjac Thank you for providing the details!

I finally found some time to take a look at this. Unfortunately I don't have access to Safari browser where I could test it, but it looks like Safari is extra careful with Content Security Headers (Chrome is not following the specs here).

I believe that setting MQTT_WS_PORT in docker-compose.yml to the correct port should fix it:

    - MQTT_WS_PORT=8080



Grafolean should then be restarted to apply the settings:

$ docker-compose down

$ docker-compose up -d



I would appreciate if you have a chance to try it out, but understand if it is not possible. At the very least this issue might help someone else in the future. 🙂

**Comment by grafolean** [comment]
Thank you for checking it out! GitHub didn't include the screeneshot, but I assume that the message is similar to the previous one.

I might be able to borrow a Mac in a few days and check it live then, will post here how it goes.

**Comment by Clausjac** [comment]
I have included the screenshots here - I think there is a bit more information this time around. Perhaps even useful:

![PastedImage-1](https://user-images.githubusercontent.com/79799346/114278354-ba059300-9a2f-11eb-99e7-d3bbcc5c3b9f.png)
![Screencapture-2021-04-10_14-04-01](https://user-images.githubusercontent.com/79799346/114278358-bbcf5680-9a2f-11eb-8263-fb307f2c2ee0.png)

**Comment by grafolean** [comment]
Thank you again for reporting this! I was able to replicate it now and fix it (v0.4.20).

It turns out that non-standard ports were not being applied to Content-Security-Policy meta tags... Apparently Safari is much more strict wrt. CSP rules than Chrome and Firefox.
